features/0351-generic-json-message/README.md

```diff
@@ -0,0 +1,130 @@
+
+Note: if we refer to a sender-agent or recipient-agent, and an implementation doesn't have one, the relevant behaviour would instead be in the sender/recipient.
+
+With JSON messaging, custom agents can communicate domain-specific information or implement domain-specific protocols without breaking compatibility with the wider DIDComm standards. Agents can also [relay](/concepts/0046-mediators-and-relays/README.md) messages between entities that are not DIDComm-enabled, allowing a wide range of communication protocols to use DIDComm to transport their messages.
+
+This protocol does not define the format of the content of a message, except that it be an [embedded attachment](/concepts/0017-attachments/README.md). Specific implementations may specify the exact format they expect. To support this flexibility, this protocol defines that every message must contain a `data-type` field, so a sender can indicate the correct format which the recipient should use to parse a message.
+
+#### Threading
+
+This protocol supports but does not require the use of the `~thread` decorator to provide message threading.
+
+## Reference
+
+#### DIDComm Message Format
+
+This protocol defines one DIDComm message format, of type `https://didcomm.org/json/1.0/message`. A `message` only requires (in addition to the standard contents of DIDComm messages) a `data-type` field and some number of JSON attachments, under the optional fields `data`, `data~attach`, and `~attach`. 
+
+The `"data-type"` field indicates the type of data contained under the attachments.
+
+The attachment fields are all optional, though there should be at least one.
+
+```json
+{
+  "@id": "123456780",
+  "@type": "https://didcomm.org/json/1.0/message",
+  "purpose": "",
+  "data-type": "",
+  "data": {},
+  "data~attach": {},
+  "~attach": []
+}
+```
+
+##### `purpose`
+
+The `purpose` of a message is a machine-readable string value which describes the *semantics* of a message - the role it plays within a protocol implemented using this RFC, for example.
+
+The purpose is one mechanism for determining which recipient(s) should be sent a message:
+
+The recipient registers on a particular purpose value with the recipient-agent, in which case the recipient-agent must relay any message with that value to the recipient. Multiple recipients can register with a single recipient-agent on a single purpose value, and a recipient can register with multiple recipient-agents.
+
+##### `data-type`
+
+The sender must provide a `data-type` identification string constant, which indicates, for the recipient, the format of the data and how it is to be processed. 
```

Comment:
I see the companion RFC uses `http/request`. Are you supplying guidance in the generic RFC on formatting of this value?